src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -456,7 +456,40 @@ public static string GetPlatformApiLevel (string platformApiLevelName)
 
 		public static string GetLibraryImportDirectoryNameForAssembly (string assemblyIdentName)
 		{
-			return string.Concat (new MD2Managed ().ComputeHash (Encoding.UTF8.GetBytes (assemblyIdentName)).Take (5).Select (b => b.ToString ("X02")));
+			if (assemblyIdentityMap == null) {
+				throw new Exception ("Mapping was not loaded!");
+			}
+			if (assemblyIdentityMap.ContainsKey (assemblyIdentName)) {
+				return assemblyIdentityMap [assemblyIdentName];
+			}
+			string id = string.Format ("{0}", assemblyIdentityMap.Count + 1);
+			assemblyIdentityMap.Add (assemblyIdentName, id);
+			return id;
+		}
+
+		static Dictionary<string, string> assemblyIdentityMap;
```

Comment:
In retrospect, it "needs" to be static in order to allow `GetLibraryImportDirectoryNameForAssembly()` to work at all.

I'm still not sure that's a good idea, though. Perhaps we should introduce a *new* type, and make `assemblyIdentityMap` and `GetLibraryImportDirectoryNameForAssembly()` *instance* members on that new type? The problem with such an approach is that all tasks that require the map would need to create an instance of this type instead of sharing a static instance, but that approach would certainly alleviate my concerns about building multiple projects -- possibly concurrently! -- in the same process over time.